src/shared/parameter/generation_scripts/cpp_header_config.py

```diff
@@ -0,0 +1,226 @@
+        self.is_top_level_config = is_top_level_config
+        self.parameters: List[CppParameter] = list()
+        self.configs: List[CppHeaderConfig] = list()
+
+    def add_parameter(self, parameter: CppParameter):
+        self.parameters.append(parameter)
+
+    def get_parameters(self) -> List[CppParameter]:
+        return self.parameters.copy()
+
+    def include_config(self, config: CppHeaderConfig):
+        self.configs.append(config)
+
+    @staticmethod
+    def join_with_tabs(
+        str: str, iterable: Iterable, num_tabs: int, half_tab: bool = False
+    ) -> str:
+        tab_size = HALF_TAB_SIZE if half_tab else TAB_SIZE
+        return (str + num_tabs * tab_size * " ").join(iterable)
+
+    @property
+    def dependency_graph(self):
+        return self.__dependency_graph
+
+    @dependency_graph.setter
+    def dependency_graph(self, dependency_graph: nx.DiGraph):
+        self.__dependency_graph = dependency_graph
+
+        self.dependency_graph_topological_order_configs = [
+            self.dependency_graph.nodes[node]["config"]
+            for node in list(reversed(list(nx.topological_sort(self.dependency_graph))))
+        ]
+
+        self.included_config_command_line_arg_entries = []
+        self.included_config_load_command_line_args_into_config_contents = []
+        for config in self.configs:
+            self.dfs_helper(config, "", "")
+
+    def dfs_helper(
+        self, config: CppHeaderConfig, arg_prefix: str, load_dependency: str
+    ):
+        """A depth first search helper for adding the necessary prefix to accessing and setting
+        parameters of included configs in loadFromCommmandLineArguments function
+
+        :param config: the current CppHeaderConfig object
+        :param arg_prefix: the prefix for accessing the arg struct
+        :param load_depndency: the prefix for accessing the actual parameter
+        """
+        arg_prefix = (
+            to_snake_case(config.config_name)
+            if not arg_prefix
+            else arg_prefix + "." + to_snake_case(config.config_name)
+        )
+        load_dependency = load_dependency + "getMutable{config_name}()->".format(
+            config_name=config.config_name
+        )
+
+        mutable_param_gen = (
+            param for param in config.parameters if not param.is_constant
+        )
+        for param in mutable_param_gen:
+            self.included_config_command_line_arg_entries.append(
+                param.command_line_option_entry_with_prefix(arg_prefix + ".")
+            )
+            self.included_config_load_command_line_args_into_config_contents.append(
+                param.load_command_line_arg_into_config_with_dependencies(
+                    load_dependency, arg_prefix + "."
+                )
+            )
+
+        # top level config has access to all configs, so no need to recursively add options
+        # if not self.is_top_level_config:
```

Comment:
I can't tell if this is commented code that should be removed or a part of the above comment.
If it's part of the above comment, remove the `:` at the end or use ``` to block comment.